src/Sentry/Internal/MainExceptionProcessor.cs

```diff
@@ -80,18 +79,34 @@ public void Process(Exception exception, SentryEvent sentryEvent)
             }
         }
 
-        internal IEnumerable<SentryException> CreateSentryException(Exception exception)
+        internal List<SentryException> CreateSentryExceptions(Exception exception)
         {
-            return exception.EnumerateChainedExceptions(_options)
-                .Select(BuildSentryException);
+            var exceptions = exception
+                .EnumerateChainedExceptions(_options)
+                .Select(BuildSentryException)
+                .ToList();
+
+            // If we've filtered out the aggregate exception, we'll need to copy over details from it.
+            if (exception is AggregateException && !_options.KeepAggregateException)
```

Comment:
it feels weird that now we have "is AggregateException" and KeepAggregateException logic in two places. both EnumerateChainedExceptions and here. can we refactor to have them co-located?

Reply:
The logic is different between those two.
- In `EnumerateChainedExceptions`, we still unwind aggregate exceptions regardless of the option.  The option only controls whether we include any aggregate exceptions in the result.
- In `CreateSentryExceptions`, we're specifically looking at the root exception because that's where the details we need to copy are.  The option just tells us that we can expect it to be filtered.

So they're not really able to be refactored together.